src/api.h

```diff
@@ -39,25 +34,325 @@ AWS_CRT_API const char *aws_crt_error_debug_str(int);
+AWS_CRT_API void aws_crt_event_loop_group_options_release(aws_crt_event_loop_group_options *options);
+AWS_CRT_API void aws_crt_event_loop_group_options_set_max_threads(
+    aws_crt_event_loop_group_options *options,
+    uint16_t max_threads);
+AWS_CRT_API aws_crt_event_loop_group *aws_crt_event_loop_group_new(aws_crt_event_loop_group_options *options);
+AWS_CRT_API void aws_crt_event_loop_group_release(aws_crt_event_loop_group *elg);
+
+/* Input stream */
+typedef struct aws_input_stream aws_crt_input_stream;
+typedef struct _aws_crt_input_stream_options aws_crt_input_stream_options;
+typedef struct aws_stream_status aws_crt_stream_status;
+typedef int(aws_crt_input_stream_seek_fn)(void *user_data, int64_t offset, int seek_basis);
+typedef int(aws_crt_input_stream_read_fn)(void *user_data, uint8_t *dest, size_t dest_length);
+typedef int(aws_crt_input_stream_get_status_fn)(void *user_data, aws_crt_stream_status *status);
+typedef int(aws_crt_input_stream_get_length_fn)(void *user_data, int64_t *out_length);
+typedef int(aws_crt_input_stream_destroy_fn)(void *user_data);
+AWS_CRT_API aws_crt_input_stream_options *aws_crt_input_stream_options_new(void);
+AWS_CRT_API void aws_crt_input_stream_options_release(aws_crt_input_stream_options *options);
+AWS_CRT_API void aws_crt_input_stream_options_set_user_data(aws_crt_input_stream_options *options, void *user_data);
+AWS_CRT_API void aws_crt_input_stream_options_set_seek(
+    aws_crt_input_stream_options *options,
+    aws_crt_input_stream_seek_fn *seek_fn);
+AWS_CRT_API void aws_crt_input_stream_options_set_read(
+    aws_crt_input_stream_options *options,
+    aws_crt_input_stream_read_fn *read_fn);
+AWS_CRT_API void aws_crt_input_stream_options_set_get_status(
+    aws_crt_input_stream_options *options,
+    aws_crt_input_stream_get_status_fn *get_status_fn);
+AWS_CRT_API void aws_crt_input_stream_options_set_get_length(
+    aws_crt_input_stream_options *options,
+    aws_crt_input_stream_get_length_fn *get_length_fn);
+AWS_CRT_API void aws_crt_input_stream_options_set_destroy(
+    aws_crt_input_stream_options *options,
+    aws_crt_input_stream_destroy_fn *destroy_fn);
+
+AWS_CRT_API aws_crt_input_stream *aws_crt_input_stream_new(aws_crt_input_stream_options *options);
+AWS_CRT_API void aws_crt_input_stream_release(aws_crt_input_stream *input_stream);
+
+/* HTTP */
+typedef struct _aws_crt_http_headers aws_crt_http_headers;
+AWS_CRT_API aws_crt_http_headers *aws_crt_http_headers_new_from_blob(uint8_t *blob, size_t blob_length);
+AWS_CRT_API void aws_crt_http_headers_release(aws_crt_http_headers *headers);
+AWS_CRT_API void aws_crt_http_headers_to_blob(
+    aws_crt_http_headers *headers,
+    uint8_t **out_blob,
+    size_t *out_blob_length);
+
+typedef struct _aws_crt_http_message aws_crt_http_message;
+AWS_CRT_API aws_crt_http_message *aws_crt_http_message_new_from_blob(uint8_t *blob, size_t blob_length);
```

Comment:
I don't think message_new_from_blob is a good idea.
a message has: method, path, headers, body
if you're going to invent a blob format which holds all those things .... it's just too much. just have separate setters for each